packages/contractkit/src/wrappers/BaseWrapper.ts

```diff
@@ -1,36 +1,19 @@
+import { zip } from '@celo/utils/lib/src/collections'
 import BigNumber from 'bignumber.js'
+import Contract from 'web3/eth/contract'
 import { TransactionObject } from 'web3/eth/types'
 import { ContractKit } from '../kit'
 import { TxOptions } from '../utils/send-tx'
 import { TransactionResult } from '../utils/tx-result'
 
 type Method<I extends any[], O> = (...args: I) => TransactionObject<O>
 
-export abstract class BaseWrapper<T> {
+export abstract class BaseWrapper<T extends Contract> {
   constructor(protected readonly kit: ContractKit, protected readonly contract: T) {}
 
-  protected proxySend<I extends any[], O>(methodFn: Method<I, O>) {
-    return (...args: I) => this.wrapSend(methodFn(...args))
-  }
-  protected proxyCall<I extends any[], O>(methodFn: Method<I, O>) {
-    return (...args: I) => methodFn(...args).call()
-  }
-
-  protected proxyCallAndTransform<I extends any[], O, F>(
-    methodFn: Method<I, O>,
-    post: (input: O) => F
-  ) {
-    return (...args: I) =>
-      methodFn(...args)
-        .call()
-        .then(post)
-  }
-
-  protected wrapSend<O>(txo: TransactionObject<O>): CeloTransactionObject<O> {
-    return {
-      send: (options?: TxOptions) => this.kit.sendTransactionObject(txo, options),
-      txo,
-    }
+  get address(): string {
+    // TODO fix typings
```

Comment:
Is this a problem with the eth typings, or something we could do?

Reply:
it seems the `_address` property is not part of web3 typings. I extracted this code from somewhere else